### PR TITLE
Backport PR #47327 on branch 1.4.x (REGR: Fix fillna making a copy when dict was given as fill value and inplace is set)

### PR DESCRIPTION
**Pull request by simonjayhawkins**
Backport PR #47327

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
@phofl XPASS on `test_fillna_dictlike_value_duplicate_colnames`. will fix-up later or feel free to push to this branch.

**Comment by phofl** [comment]
This is interesting, do you know if this was an intended change? Otherwise we should open an issue about it

**Comment by simonjayhawkins** [comment]
> This is interesting, do you know if this was an intended change? Otherwise we should open an issue about it

not sure. As this is 1.4.x, I'm not sure array manager changes are a concern. @jbrockmendel ?

**Comment by phofl** [comment]
Sorry that wasn't clear. I meant: If this is passing on 1.4.x and failing on main, this looks like a regression on main compared to the 1.4 release

**Comment by simonjayhawkins** [comment]
yeah. can open an issue for main. for this branch, i'm sure we just skip any failing array manager tests to get ci green.

**Comment by simonjayhawkins** [comment]
if there is a problem and a fix on main, we could then maybe consider backporting for 1.4.4

File: pandas/tests/frame/methods/test_update.py [hunk 1]
```diff
@@ -146,3 +148,14 @@ def test_update_with_different_dtype(self):
 
         expected = DataFrame({"a": [1, 3], "b": [np.nan, 2], "c": ["foo", np.nan]})
         tm.assert_frame_equal(df, expected)
+
+    @td.skip_array_manager_invalid_test
```
[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/tests/frame/methods/test_update.py:152]: maybe for the two added tests. but this is the backport. so should change on main for the change to be relevant.

for test_fillna_dictlike_value_duplicate_colnames it is XPASSing so we need to skip as skip_array_manager_not_implemented xfails.

**Pull request merged**